### PR TITLE
Add rule in order to enforce the declaration of a selector

### LINKED ISSUE #551: [Feature Request] Component Selector Required
Quick feature request... I'd love it if there was a rule I could enable that would force people to always include a selector for any @Component. Currently if it's for a dialog (dynamically created component) or routed page, you don't technically need one, but it makes the DOM a bit messy.

Perhaps it could be added to the existing component-selector rule?

### PR DESCRIPTION
**Pull request by wKoza**
closes #551

### CODE REVIEW

**Comment by mgechev** [comment]
I've been thinking about this. Not sure if introducing a boolean flag will be very intuitive to developers. I'd always wonder what this `true` stands for, especially after the other enumerable properties.

I'm thinking that it's worth it to introduce a separate rule. At first, it looks like an overhead, however, the more I think about it, the better the idea sounds to me. It'll be more flexible and obvious.

**Comment by wKoza** [comment]
I'm agree, use a 'magic' boolean is never a good idea. Well, I will change that very soon.

**Comment by mgechev** [comment]
@wKoza looks good to me! Will take a further look tomorrow and merge.

File: src/enforceComponentSelectorRule.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    constructor(sourceFile: ts.SourceFile, private rule: Rule) {
+        super(sourceFile, rule.getOptions());
+    }
+
+    visitNgComponent(metadata: ComponentMetadata) {
+        if (!metadata.selector) {
+            let failureConfig: string[] = [metadata.controller.name.text];
+            failureConfig.unshift(Rule.SELECTOR_FAILURE);
+            this.generateFailure(metadata.decorator.getStart(), metadata.decorator.getWidth(), failureConfig);
+        }
```
Comment (mgechev) [inline src/enforceComponentSelectorRule.ts]: `super.visitNgComponent`

File: src/enforceComponentSelectorRule.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export class EnforceComponentSelectorValidatorWalker extends NgWalker {
+
+    constructor(sourceFile: ts.SourceFile, private rule: Rule) {
+        super(sourceFile, rule.getOptions());
+    }
+
+    visitNgComponent(metadata: ComponentMetadata) {
+        if (!metadata.selector) {
+            let failureConfig: string[] = [metadata.controller.name.text];
```
Comment (mgechev) [inline src/enforceComponentSelectorRule.ts]: Could be a constant.

**Comment by rafaelss95** [comment]
Not related to this PR, but what's the standard in repo: 2 or 4 spaces?

**Comment by mgechev** [comment]
2. We should run prettier to format everything properly.

**Pull request merged**